tremor-script/src/ast/query/raw.rs

```diff
@@ -181,6 +239,274 @@ impl<'script> Upable<'script> for OperatorDeclRaw<'script> {
+        }?;
+        let into = match self.into {
+            Some(ports) => ports.into_iter().map(|ident| ident.up(helper)).collect(),
+            None => Ok(vec!["out".into(), "err".into()]),
+        }?;
+        let subquery_decl = SubqueryDecl {
+            mid: helper.add_meta_w_name(self.start, self.end, &self.id),
+            module: helper.module.clone(),
+            id: self.id,
+            params: up_maybe_params(self.params, helper)?,
+            raw_stmts: self.subquery,
+            from,
+            into,
+        };
+        let subquery_name = subquery_decl.fqsqn(&helper.module);
+        helper
+            .subqueries
+            .insert(subquery_name, subquery_decl.clone());
+        Ok(subquery_decl)
+    }
+}
+
+/// we're forced to make this pub because of lalrpop
+#[derive(Clone, Debug, PartialEq, Serialize)]
+pub struct SubqueryStmtRaw<'script> {
+    pub(crate) start: Location,
+    pub(crate) end: Location,
+    pub(crate) id: String,
+    pub(crate) target: String,
+    pub(crate) module: Vec<IdentRaw<'script>>,
+    pub(crate) params: MaybeParams<'script>,
+}
+impl_expr!(SubqueryStmtRaw);
+
+impl<'script> SubqueryStmtRaw<'script> {
+    fn mangle_id(&self, id: &str) -> String {
+        //TTODO Remove `A`
+        format! {"A__SUBQ__{}_{}",self.id, id}
+    }
+
+    fn get_args_map<'registry>(
+        &self,
+        decl_params: &Option<HashMap<String, Value<'script>>>,
+        stmt_params: Option<HashMap<String, Value<'script>>>,
+        helper: &Helper<'script, 'registry>,
+    ) -> Result<HashMap<String, Value<'script>>> {
+        match &decl_params {
+            // No params in the declaration
+            None => Ok(HashMap::new()),
+            Some(decl_params) => {
+                match stmt_params {
+                    // No params in the creation
+                    None => Ok(decl_params.clone()),
+                    // Merge elided params from the declaration.
+                    Some(mut stmt_params) => {
+                        for (param, value) in decl_params {
+                            if !stmt_params.contains_key(param) {
+                                stmt_params.insert(param.clone(), value.clone());
+                            }
+                        }
+                        // Make sure no new params were introduced in creation
+                        if stmt_params.len() == decl_params.len() {
+                            Ok(stmt_params)
+                        } else {
+                            error_generic(self, self, &"Unknown parameter", &helper.meta)
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    fn inline_params<'registry>(
+        params: Params<'script>,
+        sq_args: &HashMap<String, Value<'script>>,
+        helper: &mut Helper<'script, 'registry>,
+    ) -> Result<HashMap<String, Value<'script>>> {
+        match params {
+            // ALLOW: Only Raw params can be inlined.
```

Comment:
This needs a bit of explanation, how do we guarantee that we never try to inline processed params? And even if we do, an error would probably be better (just to be safe) then a `unreachable` this is compile time the bit of overhead isn't that bad compared to the risk of a odd script crashing the runtime